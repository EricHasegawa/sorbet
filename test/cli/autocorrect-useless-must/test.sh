#!/bin/bash

cwd="$(pwd)"
infile="$cwd/test/cli/autocorrect-useless-must/autocorrect-useless-must.rb"

tmp="$(mktemp -d)"

(
  cp "$infile" "$tmp"

  cd "$tmp" || exit 1
  if "$cwd/main/sorbet" --silence-dev-message -a autocorrect-useless-must.rb 2>&1; then
    echo "Expected to fail!"
    exit 1
  fi

  echo
  echo --------------------------------------------------------------------------
  echo

  # Also cat the file, to make sure that the autocorrect applied
  cat autocorrect-useless-must.rb

  rm autocorrect-useless-must.rb
)

echo
echo --------------------------------------------------------------------------
echo

(
  cp "$infile" "$tmp"

  cd "$tmp" || exit 1
  if "$cwd/main/sorbet" --silence-dev-message --suggest-unsafe -a autocorrect-useless-must.rb 2>&1; then
    echo "Expected to fail!"
    exit 1
  fi

  echo
  echo --------------------------------------------------------------------------
  echo

  # Also cat the file, to make sure that the autocorrect applied
  cat autocorrect-useless-must.rb

  rm autocorrect-useless-must.rb
)

rm -r "$tmp"
