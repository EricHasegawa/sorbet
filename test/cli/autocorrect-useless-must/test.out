autocorrect-useless-must.rb:3: `T.must` called on `String("1")`, which is never `nil` https://srb.help/7015
     3 |T.must('1')
               ^^^
  Got `String("1")` originating from:
    autocorrect-useless-must.rb:3:
     3 |T.must('1')
               ^^^
  Autocorrect: Done
    autocorrect-useless-must.rb:3: Replaced with `'1'`
     3 |T.must('1')
        ^^^^^^^^^^^

autocorrect-useless-must.rb:5: `T.must` called on `String`, which is never `nil` https://srb.help/7015
     5 |T.must('1' + '1').to_i
               ^^^^^^^^^
  Got `String` originating from:
    autocorrect-useless-must.rb:5:
     5 |T.must('1' + '1').to_i
               ^^^^^^^^^
  Autocorrect: Done
    autocorrect-useless-must.rb:5: Replaced with `'1' + '1'`
     5 |T.must('1' + '1').to_i
        ^^^^^^^^^^^^^^^^^
Errors: 2

--------------------------------------------------------------------------

# typed: strict

'1'

'1' + '1'.to_i

--------------------------------------------------------------------------

autocorrect-useless-must.rb:3: `T.must` called on `String("1")`, which is never `nil` https://srb.help/7015
     3 |T.must('1')
               ^^^
  Got `String("1")` originating from:
    autocorrect-useless-must.rb:3:
     3 |T.must('1')
               ^^^
  Autocorrect: Done
    autocorrect-useless-must.rb:3: Replaced with `'1'`
     3 |T.must('1')
        ^^^^^^^^^^^

autocorrect-useless-must.rb:5: `T.must` called on `String`, which is never `nil` https://srb.help/7015
     5 |T.must('1' + '1').to_i
               ^^^^^^^^^
  Got `String` originating from:
    autocorrect-useless-must.rb:5:
     5 |T.must('1' + '1').to_i
               ^^^^^^^^^
  Autocorrect: Done
    autocorrect-useless-must.rb:5: Replaced with `'1' + '1'`
     5 |T.must('1' + '1').to_i
        ^^^^^^^^^^^^^^^^^
Errors: 2

--------------------------------------------------------------------------

# typed: strict

'1'

'1' + '1'.to_i
